forte/data/data_store.py

```diff
@@ -594,16 +666,90 @@ def _default_attributes_for_type(self, type_name: str) -> List:
         return attr_list
 
+    def _new_entry_types(
+        self, type_name: str, attributes: Optional[Set[Tuple[str, str]]] = None
+    ) -> Dict[str, Tuple]:
+        r"""This function takes a fully qualified ``type_name`` class name
+        and creates a dictionary where the key is attribute of the entry
+        and value is the type information of that attribute. For example,
+
+        .. code-block:: python
+
+            type_dict =  {
+                        "document_class": (list, (str,)),
+                        "sentiment": (dict, (str, float)),
+                        "classifications": (FDict, (str, Classification))
+                    }
+
+        For each attribute, the type information is represented by a tuple
+        of two elements. The first element is the unsubscripted version of
+        the attribute's type and the second element is the type arguments
+        for the same. The `type_dict` is used to populate the type
+        information for attributes of an entry specified by ``type_name``
+        in `_type_attributes`.
+
+
+        Args:
+            type_name: A fully qualified name of an entry class.
+            attributes: This argument is used when parsing ontology
+                files. The entries in the set are a tuples of two
+                elements.
+
+                .. code-block:: python
+
+                    attributes =  {
+                                ('passage_id', 'str'),
+                                ('author', 'str')
+                            }
+        """
+        type_dict = {}
+        attr_class: Any
+        attr_args: Tuple
+
+        if attributes:
+            for attr, type_val in attributes:
+                # the type_dict only stores the type of each
+                # attribute class. When attributes and their
+                # types are defined in ontology files, these
+                # values are stored in attr_args. attr_class
+                # is empty in this case and has a value of
+                # None. But to maintain the consistency of
+                # type_dict, we only store the type of every
+                # value, even None.
+                attr_class = type(None)
+                attr_args = tuple([get_class(type_val)])
+                type_dict[attr] = tuple([attr_class, attr_args])
+
+        else:
+            attr_fields: Dict = self._get_entry_attributes_by_class(type_name)
+            for attr_name, attr_info in attr_fields.items():
+                attr_class = get_origin(attr_info.type)
+                attr_args = get_args(attr_info.type)
+
+                # Prior to Python 3.7, typing.List and typing.Dict
+                # is not converted to primitive forms of list and
+                # dict. We handle them separately here
+                if attr_class == Dict:
+                    attr_class = dict
+                if attr_class == List:
+                    attr_class = list
```

Comment:
I am not confident in this function. Are we only handling these types here? What if we are dealing with something new, like `Set`? How can we make sure this function is robust?

Reply:
So this is an issue caused by the change in the typing module from python 3.6 to python 3.7. Another workaround for this is that we can let the the output `get_origin` be stored in any way depending on the python version and just handle this in the test cases. This is similar to a suggestion made [here](https://github.com/ilevkivskyi/typing_inspect/issues/27)

Reply:
But how can we make sure we we list all these cases, how about `Set`?

Reply:
I have added a more general solution in the updated version of this PR which does not require modifying test cases